src/librustc_front/hir.rs

```diff
@@ -1226,7 +1248,7 @@ pub enum Item_ {
     /// An enum definition, e.g. `enum Foo<A, B> {C<A>, D<B>}`
     ItemEnum(EnumDef, Generics),
     /// A struct definition, e.g. `struct Foo<A> {x: A}`
-    ItemStruct(P<StructDef>, Generics),
+    ItemStruct(P<VariantData>, Generics),
```

Comment:
4-5 pointers.
I thought about inlining it but wasn't sure what's the general policy.